mycore-iview2/src/main/java/org/mycore/iview2/frontend/MCRPDFTools.java

```diff
@@ -49,14 +54,39 @@
         this.pngTools = new MCRPNGTools();
     }
 
-    static BufferedImage getThumbnail(Path pdfFile, int thumbnailSize, boolean centered) throws IOException {
+    /**
+     * The old method did not take the thumbnail size into account, if centered =
+     * false;
+     * 
+     * @see getThumbnail(int thumbnailSize, Path pdfFile, boolean centered)
+     */
+    @Deprecated
+    public static BufferedImage getThumbnail(Path pdfFile, int thumbnailSize, boolean centered) throws IOException {
+        return getThumbnail(-1, pdfFile, false);
+    }
+
+    /**
+     * This method returns a Buffered Image as thumbnail if an initial page was set,
+     * it will be return - if not the first page
+     * 
+     * @param thumbnailSize - the size: size = max(width, height) 
+     *                        a size &lt; 0 will return the original size and centered parameter will be ignored
+     * @param pdfFile       - the file from which the thumbnail will be taken
+     * @param centered      - if true, a square (thumbnail with same width and
+     *                      height) will be returned
+     * @return a BufferedImage as thumbnail
+     * 
+     * @throws IOException
+     */
+    public static BufferedImage getThumbnail(int thumbnailSize, Path pdfFile, boolean centered) throws IOException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'getThumbnail(int, Path, boolean)' has an NPath complexity of 417](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5425159)